File: src/apps/blueprint-gym/server/Domain/BlueprintGym.Domain.ExerciseTracker/Models/Exercise.cs
using System.Collections.Generic;
using BlueprintGym.Domain.Constants.Enums;
using BlueprintGym.Domain.Core.Cosmos;
using BlueprintGym.Domain.Core.Interfaces;
using BlueprintGym.Domain.Core.Models;
using Microsoft.Azure.Cosmos;

namespace BlueprintGym.Domain.ExerciseTracker.Models
{
  public class Exercise : CosmosEntity, ICosmosEntity
  {
    public override PartitionKey PK => new PartitionKey($"{this.GetType().Name}-{this.ExerciseId}");
    public string ExerciseId => this.Id;
    public string ExerciseName { get; set; }
    public string Description { get; set; }
    public IEnumerable<string> MusclesWorked { get; set; }
    public IEnumerable<string> ExerciseLabels { get; set; }
    public ExerciseState State { get; set; }
    public FitnessDifficulty Difficulty { get; set; }
    public MemberRef Author { get; set; }
    public ExerciseLink ParentVariationExercise { get; set; }
  }
}

File: apps/shared/go/go-sample/fitness-tracker/models/exercise.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type AllExercises struct {
	ID           primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	ExerciseRefs []ExerciseRef
}

type Exercise struct {
	ID                      primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	ExerciseName            string
	Description             string
	MusclesWorked           []string
	Labels                  []string
	State                   ExerciseState
	Difficulty              FitnessDifficulty
	Author                  MemberLink
	ParentVariationExercise ExerciseLink
}

type ExerciseRef struct {
	ExerciseId         string
	ExerciseName       string
	DescriptionSnippet string
	MusclesWorked      []string
	State              ExerciseState
	Difficulty         FitnessDifficulty
}
